README.rst

```diff
@@ -0,0 +1,329 @@
+============================================
+Batch Data Processing in CDAP using Workflow
+============================================
+
+The Workflow system in `Cask Data Application Platform (CDAP) <http://cdap.io>`__
+allows specifying, executing, scheduling, and monitoring complex series of jobs
+and tasks. In this guide, you will learn how it can facilitate the execution of
+`MapReduce <http://research.google.com/archive/mapreduce.html>`__
+programs in parallel based on the evaluation of boolean predicates.
+
+What You Will Build
+===================
+
+This guide will take you through building a
+`CDAP application <http://docs.cdap.io/cdap/current/en/developers-manual/building-blocks/applications.html>`__
+that uses ingested raw purchase events (of the form '``<name> bought <n> <item>s for $<price>``') to compute
+the total purchases made by a user along with the total purchases made for a particular product.
+
+You will:
+
+- Build ``PurchaseEventParser``, a
+  `MapReduce program <http://docs.cdap.io/cdap/current/en/developers-manual/building-blocks/mapreduce-programs.html>`__
+  to parse the raw purchase events and create ``Purchase`` objects from them;
+- Build ``PurchaseCounterByUser``, a
+  `MapReduce program <http://docs.cdap.io/cdap/current/en/developers-manual/building-blocks/mapreduce-programs.html>`__
+  to count the purchases made per user;
+- Build ``PurchaseCounterByProduct``, a
+  `MapReduce program <http://docs.cdap.io/cdap/current/en/developers-manual/building-blocks/mapreduce-programs.html>`__
+  to count the purchases made per product;
+- Build ``PurchaseWorkflow``, a
+  `Workflow program <http://docs.cdap.io/cdap/current/en/developers-manual/building-blocks/workflows.html>`__
+  which will first execute the MapReduce program ``PurchaseEventParser``. If the predicate ``PurchaseEventVerifier``
+  evaluates to true, the workflow will in parallel execute the MapReduce program ``PurchaseCounterByUser`` and
```

Comment:
If the predicate PEV, which uses the MapReduce counters emitted by the PEP to determine data quality, evaluates to true